frontend/src/components/VideoCardPage/VideoCardPage.js

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import Grid from '@material-ui/core/Grid';
+import TopBar from '../TopBar/TopBar';
+import UploadModal from '../UploadModal/UploadModal';
+import './styles.css';
+import VideoCardsByDate from '../VideoCardsByDate/VideoCardsByDate';
+import NavDrawer from '../NavDrawer/NavDrawer';
+import DeleteConfirmationDialog from '../DeleteConfirmationDialog/DeleteConfirmationDialog';
+import LoadingSpinner from '../LoadingSpinner/LoadingSpinner';
+
+const VideoCardPage = ({
+  renderSnackbars,
+  title,
+  size,
+  showNavDrawer,
+  videosInformation,
+  showUploadModal,
+  showDeletionDialog,
+  selectedCardIds,
+  iconsToShow,
+  iconsToShowOnSelected,
+  handleIconsClick,
+  handleIconsClickOnSelected,
+  sortedVideoDates,
+  handleSelect,
+  handleDateSelect,
+  showDownloadInProgress,
+  handleActionIconClick,
+  toggleNavDrawer,
+  toggleUploadModal,
+  handleVideoDeletion,
+  toggleDeletionDialog,
+  deleteForever,
+  children,
+  isLoading,
+}) => (
+  <>
+    {renderSnackbars()}
+    <Grid
+      className="root"
+      style={{
+        height: Object.keys(videosInformation).length === 0 ? '100vh' : 'auto',
+      }}
+      container
+      direction="column"
+    >
+      <NavDrawer
+        open={showNavDrawer}
+        onOpen={toggleNavDrawer}
+        onClose={toggleNavDrawer}
+        spaceTaken={size}
+      />
+      <UploadModal show={showUploadModal} onClose={toggleUploadModal} />
```

Comment:
Move this to Library page component, since it won't be used in RecyclingBinPage. Also, I found a bug when I upload two videos and change one video title, then the other one disappears from modal, since modal wasn't changed, I think that this bug might be fixed if you would move the UploadModal to LibraryPage

Reply:
Don